File: app/src/main/java/com/erebor/tomkins/pos/data/model/EncryptedSha1Field.java
package com.erebor.tomkins.pos.data.model;

import android.util.Log;

import androidx.annotation.NonNull;

import com.erebor.tomkins.pos.factory.SecurityEncryptionFactory;
import com.erebor.tomkins.pos.tools.SecurityEncryption;

public class EncryptedSha1Field {

    private SecurityEncryption encryption;
    private String encrypted;

    public EncryptedSha1Field() {
        this.encryption = SecurityEncryptionFactory.getInstance().getEncryption();
    }

    public EncryptedSha1Field(@NonNull String value) {
        this.encryption = SecurityEncryptionFactory.getInstance().getEncryption();
        this.encrypted = encrypt(value);
    }
    public String getValue() {
        try {
            return encrypted;
        } catch (Exception e) {
            return null;
        }
    }

    public void setValue(String value) {
        this.encrypted = value;
    }

    private String encrypt(@NonNull String value) {
        try {
            String encrypted = encryption.sh1(value);
            Log.d("EncryptedSha1Field", "Encrypting '" + value + "' --> '" + encrypted+"'");
            return encrypted;
        } catch (Exception e) {
            return null;
        }
    }

}

File: app/src/main/java/com/erebor/tomkins/pos/data/model/EncryptedSha1Field.kt
package com.erebor.tomkins.pos.data.model

import android.util.Log

import com.erebor.tomkins.pos.factory.SecurityEncryptionFactory
import com.erebor.tomkins.pos.tools.SecurityEncryption

class EncryptedSha1Field {

    private var encryption: SecurityEncryption? = null
    var value: String? = null
        get() {
            try {
                return field
            } catch (e: Exception) {
                return null
            }

        }

    constructor() {
        this.encryption = SecurityEncryptionFactory.instance.encryption
    }

    constructor(value: String) {
        this.encryption = SecurityEncryptionFactory.instance.encryption
        this.value = encrypt(value)
    }

    private fun encrypt(value: String): String? {
        try {
            val encrypted = encryption!!.sh1(value)
            Log.d("EncryptedSha1Field", "Encrypting '$value' --> '$encrypted'")
            return encrypted
        } catch (e: Exception) {
            return null
        }

    }

}
